packages/keystatic/src/form/fields/document/DocumentEditor/component-blocks/chromeful-element.tsx

```diff
@@ -167,9 +153,14 @@ function DefaultToolbarWithChrome({
     <NotEditable>
       <Flex direction="column" gap="medium">
         <Flex alignItems="center" gap="regular" style={{ userSelect: 'none' }}>
-          <ActionButton onPress={() => onShowEditMode()}>
-            {stringFormatter.format('edit')}
-          </ActionButton>
+          <DialogTrigger>
+            <ActionButton>{stringFormatter.format('edit')}</ActionButton>
+            {close => (
+              <Dialog onDismiss={close}>
```

Comment:
Assuming the dialog rendering moves into `ChromefulComponentBlockElement`, could you also add a heading to the modal, probably like `Edit {componentBlock.label}`?